### PR TITLE
SI-7767 avoid rejecting Scaladoc comments in early initializers

### PR DESCRIPTION
**Pull request by huitseeker**
review by @retronym

### CODE REVIEW

**Comment by retronym** [comment]
I've submitted a PR to this PR with a test case: https://github.com/huitseeker/scala/pull/1

**Comment by huitseeker** [comment]
@retronym thanks !

File: test/scaladoc/run/t7767.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import scala.tools.nsc.doc.model._
+import scala.tools.partest.ScaladocModelTest
+
+object Test extends ScaladocModelTest {
+
+  // This caused an infinite recursion in method inline() in CommentFactory.scala
```
Comment (retronym) [inline test/scaladoc/run/t7767.scala]: Ooops, I copy/pasted this an irrelevant comment. @huitseeker could you please remove this and `push -f`?

File: test/scaladoc/run/t7767.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  override def code = """
+      class Docable extends { /**Doc*/ val foo = 0 } with AnyRef
+  """
+
+  // no need for special settings
+  def scaladocSettings = ""
+
+  def testModel(rootPackage: Package) = {
+    import access._
+    // if it doesn't hang, the test is passed
```
Comment (retronym) [inline test/scaladoc/run/t7767.scala]: And this one.

**Comment by retronym** [comment]
LGTM

**Pull request merged**